pkg/index/index.go

```diff
@@ -14,8 +15,29 @@ func Index(arch, repo string) (*apk.APKIndex, error) {
 	var rc io.ReadCloser
 	if strings.HasPrefix(repo, "http://") || strings.HasPrefix(repo, "https://") {
 		url := fmt.Sprintf("%s/%s/APKINDEX.tar.gz", repo, arch)
-		resp, err := http.Get(url) //nolint:gosec
+		// The package repository URL for non public repositories is private and requires authentication
+		// If packages.wolfi.dev is in the URL then we can skip adding auth as it is a public repository
+		req, err := http.NewRequest("GET", url, nil)
 		if err != nil {
+			fmt.Printf("Failed to create request for APKINDEX URL %s: %v\n", url, err)
+			return nil, err
+		}
+		if !strings.Contains(url, "packages.wolfi.dev") {
+			httpAuthEnvVariableValue := os.Getenv("HTTP_AUTH")
+			if httpAuthEnvVariableValue == "" {
+				fmt.Printf("HTTP_AUTH environment variable is not set and is required for querying non public " +
+					"package repositories. Use `chainctl auth token --audience apk.cgr.dev` to get token required.\n")
+				return nil, err
+			}
+			encodedAuth := base64.StdEncoding.EncodeToString([]byte("user:" + httpAuthEnvVariableValue))
+			req.Header.Set("Authorization", "Basic "+encodedAuth)
```

Comment:
In apko's auth package there's a default Authenticator that has this behavior. We shouldn't have to set the header ourselves each time

Reply:
ack